File: set_1/arrays/python/f7cfcd2c-54dd-4cc8-80cf-6d4b42f2e58d.py
# Solution 1:

def even_odd_rearranger(arr):
    arr_len = len(arr)
    temp_arr = arr_len*[None]
    index = 0
    
    for i in range(arr_len):
        if(arr[i]%2 ==0):
            temp_arr[index] = arr[i]
            index += 1
            
    for i in range(arr_len):
        if(arr[i]%2 != 0):
            temp_arr[index] = arr[i]
            index += 1
    return temp_arr

arr1 = [8, 10, 11, 17, 29, 31, 32, 35, 38, 43]
print(f"The input array --> {arr1}")
res1 = even_odd_rearranger(arr1)
print(f"The array after even-odd re-arrangement --> {res1}")


# Solution 2:

def even_odd_rearranger2(arr):
    arr_len = len(arr)
    i=-1
    j=0
    while j != arr_len:
        if arr[j] % 2 == 0:
            i += 1
            temp = arr[i]
            arr[i] = arr[j]
            arr[j] = temp
        j += 1
    return arr
    
arr2 = [8, 10, 23, 25, 28, 29, 31, 36, 44, 48]
print(f"The input array --> {arr2}")
res2 = even_odd_rearranger2(arr2)
print(f"The array after even-odd re-arrangement --> {res2}")

File: set_1/arrays/javascript/f7cfcd2c-54dd-4cc8-80cf-6d4b42f2e58d.js
// Solution 1:
const evenOddRearranger = arr => {
    let tempArr= [];
    let arrLength = arr.length;
    let index = 0;
    for (let i = 0; i < arrLength; i++)
    {
        if (arr[i] % 2 == 0)
        {
            tempArr[index] = arr[i];
            index++;
        }
    }
    for (let i = 0; i < arrLength; i++)
    {
        if (arr[i] % 2 != 0)
        {
            tempArr[index] = arr[i];
            index++;
        }
    }
    return tempArr
}
 

const arr1 = [9, 10, 16, 20, 26, 29, 30, 35, 41, 43];
console.log(`The input array --> [${arr1}]`)

const res1 = evenOddRearranger(arr1);
console.log(`The array after even-odd re-arrangement --> [${res1}]`);


// Solution 2:
const evenOddRearranger2 = arr => {
    let arrLength = arr.length;
    let [i,j] = [-1,0];
    while (j != arrLength) {
        if (arr[j] % 2 == 0) {
            i++;
            let temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        }
        j++;
    }

    return arr
}

const arr2 = [9, 18, 23, 28, 31, 32, 35, 38, 48, 49];
console.log(`The input array --> [${arr2}]`)
const res2 = evenOddRearranger2(arr2);
console.log(`The array after even-odd re-arrangement --> [${res2}]`);